homeassistant/components/motioneye/camera.py

```diff
@@ -0,0 +1,206 @@
+            ]
+        )
+
+    listen_for_new_cameras(hass, entry, camera_add)
+    return True
+
+
+class MotionEyeMjpegCamera(MjpegCamera, CoordinatorEntity):
+    """motionEye mjpeg camera."""
+
+    def __init__(
+        self,
+        config_entry_id: str,
+        username: str,
+        password: str,
+        camera: dict[str, Any],
+        client: MotionEyeClient,
+        coordinator: DataUpdateCoordinator,
+    ):
+        """Initialize a MJPEG camera."""
+        self._surveillance_username = username
+        self._surveillance_password = password
+        self._client = client
+        self._camera_id = camera[KEY_ID]
+        self._device_id = get_motioneye_device_unique_id(
+            config_entry_id, self._camera_id
+        )
+        self._unique_id = get_motioneye_entity_unique_id(
+            config_entry_id, self._camera_id, TYPE_MOTIONEYE_MJPEG_CAMERA
+        )
+        self._motion_detection_enabled: bool = camera.get(KEY_MOTION_DETECTION, False)
+        self._available = MotionEyeMjpegCamera._is_acceptable_streaming_camera(camera)
+
+        # motionEye cameras are always streaming or unavailable.
+        self.is_streaming = True
+
+        MjpegCamera.__init__(
+            self,
+            {
+                CONF_VERIFY_SSL: False,
+                **self._get_mjpeg_camera_properties_for_camera(camera),
+            },
+        )
+        CoordinatorEntity.__init__(self, coordinator)
+
+    def _get_mjpeg_camera_properties_for_camera(
+        self, camera: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Convert a motionEye camera to MjpegCamera internal properties."""
+        auth = None
+        if camera.get(KEY_STREAMING_AUTH_MODE) in [
+            HTTP_BASIC_AUTHENTICATION,
+            HTTP_DIGEST_AUTHENTICATION,
+        ]:
+            auth = camera[KEY_STREAMING_AUTH_MODE]
+
+        return {
+            CONF_NAME: camera[KEY_NAME],
+            CONF_USERNAME: self._surveillance_username if auth is not None else None,
+            CONF_PASSWORD: self._surveillance_password if auth is not None else None,
+            CONF_MJPEG_URL: self._client.get_camera_stream_url(camera) or "",
+            CONF_STILL_IMAGE_URL: self._client.get_camera_snapshot_url(camera),
+            CONF_AUTHENTICATION: auth,
+        }
+
+    def _set_mjpeg_camera_state_for_camera(self, camera: dict[str, Any]) -> None:
+        """Set the internal state to match the given camera."""
+
+        # Sets the state of the underlying (inherited) MjpegCamera based on the updated
+        # MotionEye camera dictionary.
+        properties = self._get_mjpeg_camera_properties_for_camera(camera)
+        self._name = properties[CONF_NAME]
+        self._username = properties[CONF_USERNAME]
+        self._password = properties[CONF_PASSWORD]
+        self._mjpeg_url = properties[CONF_MJPEG_URL]
+        self._still_image_url = properties[CONF_STILL_IMAGE_URL]
+        self._authentication = properties[CONF_AUTHENTICATION]
```

Comment:
Just store `properties`. Or even better, why even generate properties and not just store the camera?

Reply:
Here is how the current arrangement came about:
   1. I need to set the variables of the underlying `MjpegCamera` camera from which I inherit, so I'm not sure how I can escape this litany of `self._foo = <something>` without more hackery.
   1. I also want to pass in the correct parameters to the `MjpegCamera` constructor from `__init__()`.
   1. I don't want to  repeat the logic in both of the above places, that is in `_get_mjpeg_camera_properties_for_camera`, which figures  out what those parameters should  be. That logic expands in future PRs as options are added.

Let me know if I'm missing something here, and there's some simpler way to just use `camera` without duplicating the logic to convert `camera`  into the parameters that `MjpegCamera` requires.

Reply:
Why are you inheriting from Mjpeg camera? Can MotionEye not do RTSP streams?

Reply:
Ah so this streams the images from the cameras and converts it to MJPEG. We used to do the same. It doesn't have a way to forward the original sources? That's a pity.

Reply:
See now how properties is used. It's meh as this was never the intention of MjpegCamera to be extended but it's happened before so it's not something worth addressing now.

Reply:
Correct:
   * There's no way I can find to pass through the original stream live ...
   * ... nor any (practical) way to extract the camera credentials ...
   * ... and the code is 100% meh ;-)

Perhaps I'll find some non-obvious mechanism, but that's for another day and I think this is good enough for now.